### PR TITLE
Part5

### PR DESCRIPTION
**Pull request by PixOBC**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,43 @@
-a.out
\ No newline at end of file
+a.out
+/.idea/.gitignore
+/.idea/.name
+/.idea/misc.xml
+/.idea/modules.xml
+/.idea/PFMCPP_Project3.iml
+/.idea/vcs.xml
```
Comment (matkatmusic) [inline .gitignore]:
```
**/.idea
**/cmake-build-debug/
```

File: main.cpp [hunk 2]
```diff
@@ -414,6 +549,28 @@ int Oscillator::makeOscillatorSquareWave(int oscillatorWave)
     return oscillatorWave + squareWave;
 }
 
+std::vector<int> Oscillator::populateWaveWithData()
+{
+    int container[10] = {0, 3, 7, 7, 10, 4, 5, 6, 3, 8};
```
Comment (matkatmusic) [inline main.cpp]: don't use raw arrays.
This isn't C.
this is C++
use containers like std::array or std::vector<> and give yourself some bounds checking safety.
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -531,6 +736,53 @@ int Arpeggiator::playNoteDuration(int noteDuration)
...
+        }
+        else if(noteDuration == 0.25)
+        {
+            std::cout << noteLengths[0.25] << " beats per 4-4 bar = 16";
+        }
+        else
+        {
+            std::cout << "ERROR! Note duration not available" << std::endl;
+        }
+    }
```
Comment (matkatmusic) [inline main.cpp]: Why is this chunk of code inside these `{ }`?
- [ ] revise?

File: main.cpp [hunk 4]
```diff
@@ -570,6 +823,25 @@ Filter Synthesiser::applyFilterToSound(double cutoff)
...
 
+void Synthesiser::chordTypeSelector(std::string mode)
+{
+    std::map<std::string, std::vector<int> > chordType
+            {
+                    {"major", {0, 4, 7}},
+                    {"minor", {0, 3, 7}},
+                    {"diminished", {0, 3, 6}},
+                    {"augmented", {0, 4, 8}}
+            };
```
Comment (matkatmusic) [inline main.cpp]: Fix the indentation
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 2 above)
Comment (matkatmusic) [reply main.cpp]: You need to replace this raw array with something else
- [ ] revise

File: main.cpp [hunk 5]
```diff
@@ -491,6 +668,33 @@ int AmplitudeEnvelope::changeDryWetMix(int wetValue)
...
+    static std::uniform_int_distribution<unsigned> distribution(0, 2);
+
+
+    for(size_t i = 0; i < 20; ++i)
+    {
+        buffer.push_back(distribution(engine));
+    }
+
+    std::cout << "Buffer sample value: ";
+    for (auto c : buffer)
```
Comment (matkatmusic) [inline main.cpp]: You haven't learned about references yet in this course.
I bet you are assuming that `c` is an element in `buffer`.
it is not.   
Don't use stuff you haven't learned about in the course.
switch to a regular `for()` loop and index the elements directly.
- [ ] revise

File: main.cpp [hunk 6]
```diff
@@ -531,6 +736,53 @@ int Arpeggiator::playNoteDuration(int noteDuration)
...
+{
+    std::map<double, std::string> noteLengths
+            {
+                    {4.0, "semibreve"},
+                    {2.0, "minim"},
+                    {1.0, "crotchet"},
+                    {0.5, "quaver"},
+                    {0.25, "semiquaver"}
+            };
+
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [ ] revise

File: main.cpp (hunk 3 above)
Comment (matkatmusic) [reply main.cpp]: Nothing was revised.
Are you trying to print out this stuff inside the `for()` loop?
Why have you put all of these `if else if` statements inside a pair of `{ }`?
- [ ] revise!!

File: main.cpp (hunk 4 above)
Comment (matkatmusic) [reply main.cpp]: You did not fix the indentation
Fix the indentation
Stop relying on the auto-formatter in your IDE.
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
You didn't fix some of the things I previously marked as needing to be revised.

File: main.cpp (hunk 5 above)
Comment (PixOBC) [reply main.cpp]: I thought c was  the cotents of buffer.begin() to buffer.end()

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**